haskell.compiler.ghc902: remove at 9.0.2

sternenseemann opened the pull request:
Let's drop everyone's least favorite GHC version.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sternenseemann committed:
haskell.compiler.ghc902: remove at 9.0.2

Let's drop everyone's least favorite GHC version.

maralorn reviewed (approved):
Still think we should do something like: https://md.darmstadt.ccc.de/FZwe_M9GQwWVnYfO0U4qJA

sternenseemann commented:
Agreed, that would be preferrable.

wolfgangwalther reviewed (commented):
I think we should proceed with this now, too.

sternenseemann commented on pkgs/top-level/haskell-packages.nix:
```diff
@@ -199,11 +185,10 @@ in
           # No suitable bindists for powerpc64le
           else if stdenv.buildPlatform.isPower64 && stdenv.buildPlatform.isLittleEndian then
-            bb.packages.ghc902
+            bb.packages.ghc8107
```
Though it's doubtful whether it is actually working right now: https://github.com/NixOS/nixpkgs/pull/282591

sternenseemann replied:
No this is impossible with the GHC 9.4 make build system which we absolutely need to use. I think the gap can be filled by packaging 9.0.2 bindists, in fact this may have already happened.

wolfgangwalther reviewed (approved):
LGTM

emilazy commented:
This should go to `master` too, right? (Are there open PRs for 8.10 and 9.2, the other versions proposed for removal in 25.11?)

wolfgangwalther committed:
doc/release-notes: Only mention GHC major version

Since we're not mentioning drops of minor versions in the release notes
anyway, we also don't need to include the minor version when we drop a
whole major.

wolfgangwalther committed:
doc/release-notes: move all GHC related release notes in one place

Also remove the minor version for GHCJS, similar to the others.
